elements/pl_number_input/pl_number_input.py

```diff
@@ -248,7 +273,24 @@ def test(element_html, element_index, data):
     name = pl.get_string_attrib(element, 'answers_name')
     weight = pl.get_integer_attrib(element, 'weight', 1)
 
-    # FIXME: randomize tests
+    # Get correct answer
+    a_tru = data['correct_answers'][name]
+
+    # If correct answer is in a format generated by pl.to_json, convert it
+    # back to a standard type (otherwise, do nothing)
+    a_tru = pl.from_json(a_tru)
 
-    data['raw_submitted_answers'][name] = '%f' % data['correct_answers'][name]
-    data['partial_scores'][name] = {'score': 1, 'weight': weight}
+    result = random.choices(['correct', 'incorrect', 'invalid'], [5, 5, 1])[0]
+    if result == 'correct':
+        data['raw_submitted_answers'][name] = str(a_tru)
+        data['partial_scores'][name] = {'score': 1, 'weight': weight}
+    elif result == 'incorrect':
+        data['raw_submitted_answers'][name] = str(a_tru + random.random())
```

Comment:
will this necessarily fail? If `random.random()` returns something very close to zero we might actually succeed? Maybe use `a_tru + random.uniform(1, 10)*random.choice([-1, 1])`?

Reply:
Done.